### ISSUE TITLE
Generate the SRL for a regex expression

### ISSUE THREAD

**Issue by francislavoie** [TYVRNET/SRL#9]
This would be a killer feature for this project - being able to take existing regex expressions and output a pretty printed SRL expression for it. That would allow verification and and modification of existing regexes much, much easier.

**Comment by KarimGeiger** [comment]
Yes, that would be pretty awesome, but I'm not sure if that's such an easy task. https://regex101.com handles displaying what's happening pretty well, but if someone has the time an knowledge to do so, you're welcome :)

**Comment by Aristona** [comment]
This would be the most useful feature ever.

+1

**Comment by gbrlmza** [comment]
Hi, this may help: [https://github.com/JexCheng/regulex](url)
SRL and Regulex, are both amazing projects

**Comment by francislavoie** [comment]
https://github.com/JexCheng/regulex/issues/16

This issue would need to be solved first, regulex doesn't support pcre
(php/perl regex)

**Comment by hoshsadiq** [comment]
Could probably reuse Doctrine's lexer to create a regex lexer (potentially separate library with a dependency)

**Comment by KarimGeiger** [comment]
You guys might want to check out https://www.regextranslator.com/ - they seem to have managed to implement a RegEx to SRL converter. Pretty cool!
